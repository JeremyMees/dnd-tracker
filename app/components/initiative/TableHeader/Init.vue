<script setup lang="ts">
import { useFormKitNodeById } from '@formkit/vue'
import { reset } from '@formkit/core'

const props = defineProps<{
  label: string
  sheet: InitiativeSheet | undefined
  update: (payload: Omit<Partial<InitiativeSheet>, NotUpdatable | 'campaign'>) => Promise<void>
}>()

const { t } = useI18n()

const popoverOpen = ref<boolean>(false)

const usedTypes = computed(() => [...new Set(props.sheet?.rows.map(({ type }) => type))])

function rollAllInitiatives() {
  const selected = useFormKitNodeById<string[]>('selected').value?.value ?? []

  props.sheet?.rows
    .filter(({ type }) => selected.includes(type))
    .forEach(({ id }) => useFormKitNodeById(id, (node: FormNode) => node.input(randomRoll(20))))
}

interface QuickInitiativeForm {
  selected: string[]
  ignore: boolean
  [key: string]: any
}

async function handleSubmit(form: QuickInitiativeForm, node: FormNode): Promise<void> {
  node.clearErrors()

  try {
    if (!props.sheet) return

    const { ignore, selected, ...characters } = sanitizeForm<QuickInitiativeForm>(form)

    const rows = [...props.sheet.rows]

    for (const key in characters) {
      if (characters[key].amount !== undefined) {
        const index = rows.findIndex(({ id }) => id === key)

        if (index >= 0) {
          let init = characters[key].amount

          if (!ignore && !isNaN(characters[key].initiative)) init += characters[key].initiative

          if (rows[index]) rows[index] = { ...rows[index], initiative: Math.max(init, 0) }
        }
      }
    }

    await props.update({ rows })
    popoverOpen.value = false
  }
  catch (err: any) {
    reset('InitiativeRowInit')
    node.setErrors(t('general.error.text'))
  }
}
</script>

<template>
  <div>
    <UiPopover v-model:open="popoverOpen">
      <UiPopoverTrigger as-child>
        <button
          id="tour-10"
          class="flex items-center gap-x-1 w-fit rounded-lg hover:bg-warning/50 hover:text-foreground hover:-mx-2 hover:-my-1 hover:px-2 hover:py-1 transition-all duration-300"
        >
          {{ label }}
          <Icon
            name="tabler:bolt-filled"
            class="text-warning"
          />
        </button>
      </UiPopoverTrigger>
      <UiPopoverContent>
        <UiPopoverHeader>
          <UiPopoverTitle>
            {{ $t('components.initiativeTableHeader.initiative.title') }}
          </UiPopoverTitle>
        </UiPopoverHeader>
        <FormKit
          id="form"
          type="form"
          :actions="false"
          @submit="handleSubmit"
        >
          <div class="flex flex-col">
            <FormKit
              v-for="row in sheet?.rows || []"
              :key="row.id"
              :name="row.id.toString()"
              type="group"
            >
              <div class="flex gap-x-2 items-end">
                <FormKit
                  :id="row.id"
                  name="amount"
                  :label="row.name"
                  validation="between:0,50|number"
                  type="number"
                  number
                  outer-class="grow"
                  suffix-icon="tabler:hexagon"
                  @suffix-icon-click="(node: FormNode) => node.input(randomRoll(20))"
                />
                <FormKit
                  name="initiative"
                  label="MOD"
                  validation="between:-20,20|number"
                  type="number"
                  number
                  min="-20"
                  max="20"
                  outer-class="w-[100px]"
                />
              </div>
            </FormKit>
          </div>
          <div class="py-4">
            <FormKit
              id="selected"
              name="selected"
              type="checkbox"
              :value="usedTypes"
              :label="$t('components.initiativeTableHeader.initiative.select')"
              :options="usedTypes.map((type) => ({
                label: $t(`general.${type}`),
                value: type,
              }))"
              options-class="flex flex-wrap gap-y-2 gap-x-6"
            />
            <FormKit
              name="ignore"
              type="toggle"
              outer-class="!mb-0"
              wrapper-class="!mb-0"
              :label="$t('components.initiativeTableHeader.initiative.ignore')"
            />
          </div>
          <div class="flex gap-2 items-center">
            <FormKit
              :aria-label="$t('components.initiativeTableHeader.initiative.roll')"
              type="button"
              input-class="$reset !mb-0 btn-primary"
              outer-class="!mb-0"
              @click="rollAllInitiatives"
            >
              {{ $t('components.initiativeTableHeader.initiative.roll') }}
            </FormKit>
            <FormKit
              :aria-label="$t('actions.update')"
              outer-class="!mb-0 grow"
              type="submit"
            >
              {{ $t('actions.update') }}
            </FormKit>
          </div>
        </FormKit>
      </UiPopoverContent>
    </UiPopover>
  </div>
</template>
