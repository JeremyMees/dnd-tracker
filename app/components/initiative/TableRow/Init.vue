<script setup lang="ts">
import { reset } from '@formkit/core'
import { INITIATIVE_SHEET } from '~~/constants/provide-keys'

const props = defineProps<{ item: InitiativeSheetRow }>()

const { sheet, update } = validateInject(INITIATIVE_SHEET)

const { t } = useI18n()

const popoverOpen = ref<boolean>(false)

const currentIndex = computed(() => props.item.index)

const canGoUp = computed(() => {
  if (!sheet.value) return false

  const isNotFirst = currentIndex.value > 0
  const isSameInitiative = sheet.value.rows[currentIndex.value - 1]?.initiative === props.item.initiative

  return isNotFirst && isSameInitiative
})

const canGoDown = computed(() => {
  if (!sheet.value) return false

  const isNotLast = currentIndex.value < sheet.value.rows.length - 1
  const isSameInitiative = sheet.value.rows[currentIndex.value + 1]?.initiative === props.item.initiative

  return isNotLast && isSameInitiative
})

async function moveRow(up: boolean): Promise<void> {
  if (!sheet.value) return

  const rows = [...sheet.value.rows]
  const index = props.item.index
  const targetIndex = up ? index - 1 : index + 1

  if (
    (up && index <= 0)
    || (!up && index >= rows.length - 1)
    || !rows[index]
    || !rows[targetIndex]
  ) return

  // Just change the index properties without swapping the array positions
  rows[index] = { ...rows[index], index: targetIndex }
  rows[targetIndex] = { ...rows[targetIndex], index: index }

  // When moving up, update all following indexes to maintain sequence
  if (up) {
    for (let i = index + 1; i < rows.length; i++) {
      if (rows[i]) rows[i] = { ...rows[i], index: i } as InitiativeSheetRow
    }
  }

  await update({ rows: [...rows].sort((a, b) => a.index - b.index) })
}

interface InitiativeForm { initiative: number, modifier?: number }

async function handleSubmit(form: InitiativeForm, node: FormNode): Promise<void> {
  node.clearErrors()

  try {
    if (!sheet.value) return

    const { initiative, modifier } = sanitizeForm<InitiativeForm>(form)

    const index = getCurrentRowIndex(sheet.value, props.item.id)
    const rows = [...sheet.value.rows]

    if (index === -1 || !rows[index]) return

    rows[index] = {
      ...rows[index],
      initiative: Math.max(0, initiative + (modifier ?? 0)),
    }

    await update({ rows })
    popoverOpen.value = false
  }
  catch (err: any) {
    reset('InitiativeRowInit')
    node.setErrors(t('general.error.text'))
  }
}
</script>

<template>
  <div class="flex gap-2 items-center text-left">
    <UiPopover v-model:open="popoverOpen">
      <UiPopoverTrigger as-child>
        <button
          data-test-trigger
          class="flex flex-col justify-center"
        >
          <Icon
            v-if="item.initiative < 0"
            data-test-empty
            name="tabler:plus"
            class="size-5 min-w-5 text-foreground/10"
            aria-hidden="true"
          />
          <span
            v-else
            data-test-initiative
          >
            {{ item.initiative }}
          </span>
        </button>
      </UiPopoverTrigger>
      <UiPopoverContent>
        <UiPopoverHeader>
          <UiPopoverTitle>
            {{ $t('components.initiativeTableModals.init') }}
          </UiPopoverTitle>
        </UiPopoverHeader>
        <FormKit
          id="InitiativeRowInit"
          type="form"
          :submit-label="$t('actions.save')"
          @submit="handleSubmit"
        >
          <FormKit
            name="initiative"
            :label="$t('components.inputs.amountLabel')"
            :value="item.initiative < 0 ? undefined : item.initiative"
            validation="required|between:0,50|number"
            type="number"
            suffix-icon="tabler:hexagon"
            number
            @suffix-icon-click="(node: FormNode) => node.input(randomRoll(20))"
          />
          <FormKit
            name="modifier"
            :label="`${$t('components.inputs.initiativeLabel')} (MODIFIER)`"
            :value="item.initiative_modifier"
            validation="between:-20,20|number"
            min="-20"
            max="20"
            type="number"
            number
          />
        </FormKit>
      </UiPopoverContent>
    </UiPopover>
    <div
      v-if="item.initiative !== null && item.initiative >= 0"
      data-test-controls
      class="flex flex-col"
    >
      <button
        v-if="canGoUp"
        v-tippy="$t('actions.moveUp')"
        data-test-up
        :aria-label="$t('actions.moveUp')"
        :class="{ 'relative top-1': canGoDown }"
        class="flex items-center"
        @click="moveRow(true)"
      >
        <Icon
          name="tabler:caret-up"
          class="size-5 min-w-5 text-tertiary"
          aria-hidden="true"
        />
      </button>
      <button
        v-if="canGoDown"
        v-tippy="$t('actions.moveDown')"
        data-test-down
        :aria-label="$t('actions.moveDown')"
        :class="{ 'relative bottom-1': canGoUp }"
        class="flex items-center"
        @click="moveRow(false)"
      >
        <Icon
          name="tabler:caret-down"
          class="size-5 min-w-5 text-tertiary"
          aria-hidden="true"
        />
      </button>
    </div>
  </div>
</template>
