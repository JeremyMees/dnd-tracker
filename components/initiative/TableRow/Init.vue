<script setup lang="ts">
import { reset } from '@formkit/core'

const props = defineProps<{
  item: InitiativeSheetRow
  sheet: InitiativeSheet | undefined
  update: (payload: Omit<Partial<InitiativeSheet>, NotUpdatable | 'campaign'>) => Promise<void>
}>()

const { t } = useI18n()

const popoverOpen = ref<boolean>(false)

const currentIndex = computed(() => props.item.index)
const canGoUp = computed(() => {
  if (!props.sheet) return false
  return currentIndex.value > 0 && props.sheet.rows[currentIndex.value - 1]?.initiative === props.item.initiative
})
const canGoDown = computed(() => {
  if (!props.sheet) return false
  return currentIndex.value < props.sheet.rows.length - 1 && props.sheet.rows[currentIndex.value + 1]?.initiative === props.item.initiative
})

async function moveRow(up: boolean): Promise<void> {
  if (!props.sheet) return

  const rows = [...props.sheet.rows]
  const index = props.item.index
  const targetIndex = up ? index - 1 : index + 1

  if (up && index <= 0) return
  if (!up && index >= rows.length - 1) return

  // Just change the index properties without swapping the array positions
  rows[index] = { ...rows[index], index: targetIndex }
  rows[targetIndex] = { ...rows[targetIndex], index: index }

  // When moving up, update all following indexes to maintain sequence
  if (up) {
    for (let i = index + 1; i < rows.length; i++) {
      rows[i] = { ...rows[i], index: i }
    }
  }

  await props.update({ rows: [...rows].sort((a, b) => a.index - b.index) })
}

interface InitiativeForm { initiative: number, modifier?: number }

async function handleSubmit(form: InitiativeForm, node: FormNode): Promise<void> {
  node.clearErrors()

  try {
    if (!props.sheet) return

    const { initiative, modifier } = sanitizeForm<InitiativeForm>(form)

    const index = getCurrentRowIndex(props.sheet, props.item.id)
    const rows = [...props.sheet.rows]

    if (index === -1) return

    rows[index] = {
      ...rows[index],
      initiative: Math.max(0, initiative + (modifier ?? 0)),
    }

    await props.update({ rows })
    popoverOpen.value = false
  }
  catch (err: any) {
    reset('InitiativeRowInit')
    node.setErrors(t('general.error.text'))
  }
}
</script>

<template>
  <div class="flex gap-2 items-center text-left">
    <UiPopover v-model:open="popoverOpen">
      <UiPopoverTrigger as-child>
        <button class="flex flex-col justify-center">
          <Icon
            v-if="item.initiative < 0"
            name="tabler:plus"
            class="size-5 min-w-5 text-foreground/10"
            aria-hidden="true"
          />
          <span v-else>{{ item.initiative }}</span>
        </button>
      </UiPopoverTrigger>
      <UiPopoverContent>
        <UiPopoverHeader>
          <UiPopoverTitle>
            {{ $t('components.initiativeTableModals.init') }}
          </UiPopoverTitle>
        </UiPopoverHeader>
        <FormKit
          id="InitiativeRowInit"
          type="form"
          :submit-label="$t('actions.save')"
          @submit="handleSubmit"
        >
          <FormKit
            name="initiative"
            :label="$t('components.inputs.amountLabel')"
            :value="item.initiative < 0 ? undefined : item.initiative"
            validation="required|between:0,50|number"
            type="number"
            suffix-icon="tabler:hexagon"
            number
            @suffix-icon-click="(node: FormNode) => node.input(randomRoll(20))"
          />
          <FormKit
            name="modifier"
            :label="`${$t('components.inputs.initiativeLabel')} (MODIFIER)`"
            :value="item.initiative_modifier"
            validation="between:-20,20|number"
            min="-20"
            max="20"
            type="number"
            number
          />
        </FormKit>
      </UiPopoverContent>
    </UiPopover>
    <div
      v-if="item.initiative !== null && item.initiative >= 0"
      class="flex flex-col"
    >
      <button
        v-if="canGoUp"
        v-tippy="$t('actions.moveUp')"
        :aria-label="$t('actions.moveUp')"
        :class="{ 'relative top-1': canGoDown }"
        class="flex items-center"
        @click="moveRow(true)"
      >
        <Icon
          name="tabler:caret-up"
          class="size-5 min-w-5 text-tertiary"
          aria-hidden="true"
        />
      </button>
      <button
        v-if="canGoDown"
        v-tippy="$t('actions.moveDown')"
        :aria-label="$t('actions.moveDown')"
        :class="{ 'relative bottom-1': canGoUp }"
        class="flex items-center"
        @click="moveRow(false)"
      >
        <Icon
          name="tabler:caret-down"
          class="size-5 min-w-5 text-tertiary"
          aria-hidden="true"
        />
      </button>
    </div>
  </div>
</template>
