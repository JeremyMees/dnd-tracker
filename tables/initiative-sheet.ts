import { createColumnHelper, type InitialTableState, type Row } from '@tanstack/vue-table'
import { actionsTable, expandButton, iconButton } from './generate-functions'

import {
  InitiativeTableRowName,
  InitiativeTableRowHealth,
  InitiativeTableRowActions,
  InitiativeTableRowConditions,
  InitiativeTableRowNotes,
  InitiativeTableRowInit,
  InitiativeTableRowAc,
  InitiativeTableRowDeathSaves,
  InitiativeTableRowConcentration,
  InitiativeTableRowModify,
} from '#components'

const columnHelper = createColumnHelper<InitiativeSheetRow>()

interface ColumnOptions {
  sheet: ComputedRef<InitiativeSheet | undefined>
  update: (payload: Omit<Partial<InitiativeSheet>, NotUpdatable>) => Promise<void>
  openQuickInitModal: () => void
}

export function generateColumns({ sheet, update, openQuickInitModal }: ColumnOptions) {
  const { t } = useI18n()

  return [
    columnHelper.display({
      enableGlobalFilter: false,
      id: 'index',
      cell: ({ row }) => h('div', { class: 'flex items-center gap-x-2 max-w-20' }, [
        h('span', { class: 'text-muted-foreground ml-1' }, row.index + 1),
        expandButton({
          content: t(`actions.${row.getIsExpanded() ? 'hide' : 'show'}`),
          expanded: row.getIsExpanded(),
          cb: () => row.toggleExpanded(),
        }),
      ]),
    }),
    columnHelper.accessor('name', {
      header: t('components.encounterTable.headers.name'),
      cell: ({ row }) => h(InitiativeTableRowName, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('initiative', {
      header: () => h('button', { class: 'flex items-center gap-x-2' }, [
        t('components.encounterTable.headers.init'),
        iconButton({
          icon: 'tabler:bolt-filled',
          content: t('components.encounterTable.quick'),
          color: 'warning',
          cb: () => openQuickInitModal(),
          disabled: !sheet.value?.rows.length,
          iconColor: 'text-warning',
        }),
      ]),
      cell: ({ row }) => h(InitiativeTableRowInit, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('health', {
      header: t('components.encounterTable.headers.health'),
      cell: ({ row }) => h(InitiativeTableRowHealth, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('ac', {
      header: t('components.encounterTable.headers.ac'),
      cell: ({ row }) => h(InitiativeTableRowAc, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.display({
      id: 'manage',
      header: t('components.encounterTable.headers.manage'),
      cell: ({ row }) => h(InitiativeTableRowActions, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('conditions', {
      header: t('components.encounterTable.headers.conditions'),
      cell: ({ row }) => h(InitiativeTableRowConditions, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('note', {
      header: t('components.encounterTable.headers.note'),
      cell: ({ row }) => h(InitiativeTableRowNotes, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('deathSaves', {
      header: t('components.encounterTable.headers.deathSaves'),
      cell: ({ row }) => h(InitiativeTableRowDeathSaves, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('concentration', {
      header: t('components.encounterTable.headers.concentration'),
      cell: ({ row }) => h(InitiativeTableRowConcentration, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.display({
      enableGlobalFilter: false,
      enableSorting: false,
      id: 'actions',
      cell: ({ row }) => h(InitiativeTableRowModify, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
  ]
}

export function expandedMarkup(row: Row<InitiativeSheetRow>) {
  return actionsTable(row.original)
}

export const initialState: InitialTableState = {}
