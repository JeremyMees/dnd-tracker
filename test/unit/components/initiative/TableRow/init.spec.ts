import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it, vi, beforeEach } from 'vitest'
import Init from '~/components/initiative/TableRow/Init.vue'
import { sheet } from '~~/test/unit/fixtures/initiative-sheet'

interface Props {
  item: InitiativeSheetRow
  sheet: InitiativeSheet | undefined
  update: (payload: Omit<Partial<InitiativeSheet>, NotUpdatable | 'campaign'>) => Promise<void>
}

const mockUpdate = vi.fn()

const props: Props = {
  item: sheet.rows[0]!,
  sheet,
  update: mockUpdate,
}

describe('Initiative table row init', async () => {
  beforeEach(() => mockUpdate.mockClear())

  it('Should match snapshot', async () => {
    const component = await mountSuspended(Init, { props })

    expect(component.html()).toMatchSnapshot()
  })

  it('Should display initiative value correctly', async () => {
    const initiative = 15
    const component = await mountSuspended(Init, {
      props: {
        ...props,
        item: { ...props.item, initiative },
      },
    })

    expect(component.get('[data-test-initiative]').text()).toBe(initiative.toString())
  })

  it('Should show plus icon when initiative is not defined', async () => {
    const component = await mountSuspended(Init, {
      props: {
        ...props,
        item: { ...props.item, initiative: -1 },
      },
    })

    expect(component.get('[data-test-empty]').isVisible()).toBeTruthy()
  })

  it('Should show up/down controls when initiative is defined', async () => {
    const component = await mountSuspended(Init, {
      props: {
        ...props,
        item: { ...props.item, initiative: 15 },
      },
    })

    expect(component.get('[data-test-controls]').isVisible()).toBeTruthy()
  })

  it('Should not show controls when initiative is not defined', async () => {
    const component = await mountSuspended(Init, {
      props: {
        ...props,
        item: { ...props.item, initiative: -1 },
      },
    })

    expect(component.find('[data-test-controls]').exists()).toBeFalsy()
  })

  it('Should enable up button when can move up', async () => {
    const component = await mountSuspended(Init, {
      props: {
        ...props,
        item: { ...props.item, initiative: 15, index: 1 },
        sheet: {
          ...sheet,
          rows: [
            { ...sheet.rows[0]!, initiative: 15, index: 0 },
            { ...props.item, initiative: 15, index: 1 },
          ],
        },
      },
    })

    expect(component.get('[data-test-up]').isVisible()).toBeTruthy()
  })

  it('Should enable down button when can move down', async () => {
    const component = await mountSuspended(Init, {
      props: {
        ...props,
        item: { ...props.item, initiative: 15, index: 0 },
        sheet: {
          ...sheet,
          rows: [
            { ...props.item, initiative: 15, index: 0 },
            { ...sheet.rows[0]!, initiative: 15, index: 1 },
          ],
        },
      },
    })

    expect(component.get('[data-test-down]').isVisible()).toBeTruthy()
  })

  it('Should move row up when clicking up button', async () => {
    const firstRow = { ...sheet.rows[0]!, initiative: 15, index: 0, id: 'row1' }
    const secondRow = { ...props.item, initiative: 15, index: 1, id: 'row2' }

    const component = await mountSuspended(Init, {
      props: {
        ...props,
        item: secondRow,
        sheet: {
          ...sheet,
          rows: [firstRow, secondRow],
        },
      },
    })

    await component.get('[data-test-up]').trigger('click')

    expect(mockUpdate).toHaveBeenCalled()

    const payload = mockUpdate.mock.calls[0]?.[0] as { rows: InitiativeSheetRow[] }
    expect(payload).toBeDefined()

    const resultRows = payload.rows
    expect(resultRows[0]?.id).toBe('row2')
    expect(resultRows[1]?.id).toBe('row1')
  })

  it('Should move row down when clicking down button', async () => {
    const firstRow = { ...props.item, initiative: 15, index: 0, id: 'row1' }
    const secondRow = { ...sheet.rows[0]!, initiative: 15, index: 1, id: 'row2' }

    const component = await mountSuspended(Init, {
      props: {
        ...props,
        item: firstRow,
        sheet: {
          ...sheet,
          rows: [firstRow, secondRow],
        },
      },
    })

    await component.get('[data-test-down]').trigger('click')

    expect(mockUpdate).toHaveBeenCalled()

    const payload = mockUpdate.mock.calls[0]?.[0] as { rows: InitiativeSheetRow[] }
    expect(payload).toBeDefined()

    const resultRows = payload.rows
    expect(resultRows[0]?.id).toBe('row2')
    expect(resultRows[1]?.id).toBe('row1')
  })

  it('Should not show up/down buttons when initiative values are different', async () => {
    const component = await mountSuspended(Init, {
      props: {
        ...props,
        item: { ...props.item, initiative: 15, index: 1 },
        sheet: {
          ...sheet,
          rows: [
            { ...sheet.rows[0]!, initiative: 20, index: 0 },
            { ...props.item, initiative: 15, index: 1 },
          ],
        },
      },
    })

    expect(component.find('[data-test-up]').exists()).toBeFalsy()
    expect(component.find('[data-test-down]').exists()).toBeFalsy()
  })
})
