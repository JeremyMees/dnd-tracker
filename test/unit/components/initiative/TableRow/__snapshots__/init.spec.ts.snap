// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Initiative table row init > Should match snapshot 1`] = `
"<div class="flex gap-2 items-center text-left"><button id="reka-popover-trigger-v-0-0-0" type="button" aria-haspopup="dialog" aria-expanded="false" aria-controls="" data-state="closed" data-test-trigger="" class="flex flex-col justify-center"><span data-test-initiative="">20</span></button>
  <!--teleport start-->
  <!--teleport end-->
  <div data-test-controls="" class="flex flex-col">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
